### PR TITLE
Make ReturnableBlockLowering common

### PR DESCRIPTION
**Pull request by pyos**
and remove special handling in JVM_IR codegen.

### CODE REVIEW

**Comment by pyos** [comment]
@max-kammerer can you have a look at this, please?

**Comment by max-kammerer** [comment]
@pyos Yes, of course, sorry for delay

File: compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/ReturnableBlockLowering.kt [hunk 1]
```diff
@@ -69,40 +69,25 @@ import org.jetbrains.kotlin.ir.visitors.IrElementTransformer
...
-private class ReturnableBlockLoweringContext(val containingDeclaration: IrDeclarationParent) {
-    var labelCnt = 0
-    val returnMap = mutableMapOf<IrReturnableBlockSymbol, (IrReturn) -> IrExpression>()
-}
-
-private class ReturnableBlockTransformer(
-    val context: CommonBackendContext
-) : IrElementTransformer<ReturnableBlockLoweringContext> {
+class ReturnableBlockLowering(val context: CommonBackendContext) : FileLoweringPass, IrElementTransformerVoidWithContext() {
+    private var labelCnt = 0
```
Comment (max-kammerer) [inline compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/ReturnableBlockLowering.kt]: Counter was file specific but now it's  module specific.

File: compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/ReturnableBlockLowering.kt [hunk 2]
```diff
@@ -69,40 +69,25 @@ import org.jetbrains.kotlin.ir.visitors.IrElementTransformer
...
+    private val returnMap = mutableMapOf<IrReturnableBlockSymbol, (IrReturn) -> IrExpression>()
```
Comment (max-kammerer) [inline compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/ReturnableBlockLowering.kt]: same here

File: compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/ReturnableBlockLowering.kt (hunk 1 above)
Comment (pyos) [reply compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/ReturnableBlockLowering.kt]: Thanks, haven't noticed that the JS backend only constructs file lowerings once, unlike the JVM one.

**Comment by max-kammerer** [comment]
Thanks!

**Pull request merged**